### PR TITLE
Add kraken2 run module

### PR DESCRIPTION
**Pull request by JoseEspinosa**
Add kraken2 run module as a [requirement](https://github.com/nf-core/viralrecon/issues/149) of `nfcore/viralrecon` DSL2 implementation.

The previous [PR](https://github.com/nf-core/modules/pull/255) adding this module was closed since the structure of the test data was changed.

To construct the `kraken2` DB for testing purposes the `fasta` file on `tests/data/genomics/sarscov2/fasta/test_genome.fasta` was used after adding the coronavirus taxonomy ID the fasta header `|kraken:taxid|2697049`

Then the kraken2 custom database was created using these commands:

``` 
DBNAME='test_kraken2_DB'
kraken2-build --download-taxonomy --db $DBNAME --use-ftp
kraken2-build --add-to-library test_genome.fasta --db $DBNAME
kraken2-build --build --db $DBNAME
```

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `<SOFTWARE>.version.txt` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
    - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd`

### CODE REVIEW

**Comment by JoseEspinosa** [comment]
I modified the header of the test `fasta` file on `tests/data/genomics/sarscov2/fasta/test_genome.fasta` since the taxonomy ID is needed by `kraken2` to build the DB (`|kraken:taxid|2697049`). However. I didn't make this change permanent to avoid possible problems with other tools that might parse the header, what do you think @drpatelh ?

File: software/functions.nf [hunk 1]
```diff
@@ -0,0 +1,59 @@
+/*
+ * -----------------------------------------------------
+ *  Utility functions used in nf-core DSL2 module files
```
Comment (drpatelh) [inline software/functions.nf:3]: Rogue file? Needs to be deleted.

File: software/kraken2/run/meta.yml [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+            e.g. [ id:'test', single_end:false ]
+    - reads:
+        type: file
+        description: |
+            List of input FastQ files of size 1 and 2 for single-end and paired-end data,
+            respectively.
+    - db:
+        type: directory
+        description: Kraken2 database
+
```
Comment (drpatelh) [inline software/kraken2/run/meta.yml:50]:
```suggestion
```

File: software/main.nf [hunk 3]
```diff
@@ -0,0 +1,52 @@
+// Import generic module functions
+include { initOptions; saveFiles; getSoftwareName } from './functions'
```
Comment (drpatelh) [inline software/main.nf:2]: Another rogue file that needs to be deleted?

File: software/meta.yml [hunk 4]
```diff
@@ -0,0 +1,81 @@
+name: kraken2_run
+description: Classifies metagenomic sequence data
+keywords:
```
Comment (drpatelh) [inline software/meta.yml:3]: Another rogue file that needs to be deleted?

**Review by drpatelh — CHANGES_REQUESTED** [review]
Yeah, leave the DB as it is for now @JoseEspinosa. Although, I think we will need to change it when we eventually add the Kraken2 DB index process 😅 We really need a workflow to generate all of this test data! Means we can use small adjustments like this and re-generate everything. Too much hassle at the moment to do it individually like @KevinMenden did 😎 !

Need to delete some files on the PR though.

**Comment by drpatelh** [comment]
Needs a few changes still here @JoseEspinosa to bring it in line with the new structure and to remove rogue files.

**Comment by JoseEspinosa** [comment]
@drpatelh I think I have addressed all the requested changes.

**Review by drpatelh — APPROVED** [review]
Thanks @JoseEspinosa !

**Pull request merged**